commcare_connect/solicitations/tables.py

```
@@ -0,0 +1,342 @@
+    badge_class = get_status_badge_class(solicitation_type)
+    badge_text = solicitation_type.upper()
+    return format_html('<div class="text-wrap">{} <span class="{}">{}</span></div>', title, badge_class, badge_text)
+
+
+def render_two_line_text(title, subtitle=None):
+    """Render two-line text content"""
+    if subtitle:
+        return format_html(
+            '<div><div class="font-medium">{}</div><div class="text-xs text-gray-500">{}</div></div>',
+            title,
+            mark_safe(subtitle) if isinstance(subtitle, str) and ("<" in subtitle and ">" in subtitle) else subtitle,
```

Ruff (0.14.3)

Use of `mark_safe` may expose cross-site scripting vulnerabilities

(S308)